lib/SchemaConnector.js

```diff
@@ -0,0 +1,249 @@
+'use strict';
+
+const DiscoveryResponse = require("./discovery/DiscoveryResponse");
+const StateRefreshResponse = require("./state/StateRefreshResponse");
+const CommandResponse = require("./state/CommandResponse");
+const AccessTokenRequest = require("./callbacks/AccessTokenRequest");
+const ErrorResponse = require('./ErrorResponse');
+const STBase = require('./STBase');
+
+module.exports = class SchemaConnector {
+
+  constructor(options = {}) {
+    this._clientId = options.clientId;
```

Comment:
Instead of using "_" to indicate private.  Try using Symbol.  Check out this example from our code.  https://git.magnapithos.net/iot/device-platform/viper/blob/master/libs/discovery-request.js#L24

Let me know if you need help understanding how this works.

Reply:
Here's more detail.  http://2ality.com/2016/01/private-data-classes.html

Reply:
:thumbsup: